### DISCUSSION TITLE
Helper method to get component from an element sourceId

### DISCUSSION THREAD

**Discussion by timkelty** [craftcms/cms#8890] [Ideas]
AFAIK, there is no helper method to get a backing component from an element sourceId.

Eg, given `section:uid-xxxx-xxxx-xxxx`, return the `Section`, or the `Volume` from `volume:uid-xxxx-xxxx-xxxx`.

Obviously it's trivial enough to split on the `:` to get the UID (like you're doing [here](https://github.com/craftcms/cms/blob/14f0fd652b7a99820f804ba6c0a0db762bf26206/src/fields/Assets.php#L909-L919)), but seems like a common enough thing that a helper would be nice.

Use-case: I ended up wanting it when registering some element actions:

```
        Event::on(
            Entry::class,
            Entry::EVENT_REGISTER_ACTIONS,
            function (RegisterElementActionsEvent $event) {
                $sourceParts = explode(':', $event->source, 2);

                if (count($sourceParts) !== 2) {
                    return;
                }

                if (!$section = Craft::$app->sections->getSectionByUid($sourceParts[1])) {
                    return;
                }

                if ($section->handle === 'orders') {
                    // …
                }

                if ($section->handle === 'creditRequests') {
                    // …
                }
            },
        );
```

**Comment by brandonkelly** [comment]
Realistically this would end up needing to be several helper methods, not just one, since source keys are element type-specific and not meant to be something that could be understood on its own.
